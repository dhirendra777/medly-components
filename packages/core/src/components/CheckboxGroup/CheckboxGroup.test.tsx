import { cleanup, fireEvent, render } from '@test-utils';
import React from 'react';
import { CheckboxGroup } from './CheckboxGroup';

const options = [
    { value: 'honda', label: 'Honda' },
    { value: 'hyundai', label: 'Hyundai' },
    {
        value: [
            { value: 'jaguar', label: 'Jaguar' },
            { value: 'landRover', label: 'Land Rover' }
        ],
        label: 'Tata'
    }
];

describe('CheckboxGroup component', () => {
    afterEach(cleanup);

    it('should render correctly with all the props', () => {
        const { container } = render(<CheckboxGroup fullWidth disabled onChange={jest.fn()} label="Cars" options={options} />);
        expect(container).toMatchSnapshot();
    });

    it('should deselect option when it is already selected', () => {
        const mockOnChange = jest.fn(),
            { getByText } = render(<CheckboxGroup values={['hyundai']} onChange={mockOnChange} label="Cars" options={options} />);
        fireEvent.click(getByText('Hyundai'));
        expect(mockOnChange).toHaveBeenCalledWith([]);
    });

    it('should select option when click on it', () => {
        const mockOnChange = jest.fn(),
            { getByText } = render(<CheckboxGroup onChange={mockOnChange} label="Cars" options={options} />);
        fireEvent.click(getByText('Hyundai'));
        expect(mockOnChange).toHaveBeenCalledWith(['hyundai']);
    });

    it('should select all when clicked on the select all option', () => {
        const mockOnChange = jest.fn(),
            { getByText } = render(
                <CheckboxGroup showSelectAll values={['hyundai']} onChange={mockOnChange} label="Cars" options={options} />
            );
        fireEvent.click(getByText('Cars'));
        expect(mockOnChange).toHaveBeenCalledWith(['hyundai', 'honda', 'jaguar', 'landRover']);
    });

    it('should unselect all when all the options are already selected', () => {
        const mockOnChange = jest.fn(),
            { getByText } = render(
                <CheckboxGroup
                    showSelectAll
                    values={['honda', 'hyundai', 'jaguar', 'landRover']}
                    onChange={mockOnChange}
                    label="Cars"
                    options={options}
                />
            );
        fireEvent.click(getByText('Cars'));
        expect(mockOnChange).toHaveBeenCalledWith([]);
    });

    describe('nested options', () => {
        it('should select option when click on it', () => {
            const mockOnChange = jest.fn(),
                { getByText } = render(<CheckboxGroup onChange={mockOnChange} label="Cars" options={options} />);
            fireEvent.click(getByText('Jaguar'));
            expect(mockOnChange).toHaveBeenCalledWith(['jaguar']);
        });

        it('should select all nested options when clicked on the group label', () => {
            const mockOnChange = jest.fn(),
                { getByText } = render(<CheckboxGroup showSelectAll onChange={mockOnChange} label="Cars" options={options} />);
            fireEvent.click(getByText('Tata'));
            expect(mockOnChange).toHaveBeenCalledWith(['jaguar', 'landRover']);
        });

        it('should unselect all when all the options are already selected', () => {
            const mockOnChange = jest.fn(),
                { getByText } = render(
                    <CheckboxGroup showSelectAll values={['jaguar', 'landRover']} onChange={mockOnChange} label="Cars" options={options} />
                );
            fireEvent.click(getByText('Tata'));
            expect(mockOnChange).toHaveBeenCalledWith([]);
        });
    });
});
