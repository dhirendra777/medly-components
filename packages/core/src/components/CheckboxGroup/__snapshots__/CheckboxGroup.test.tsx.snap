// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup component should render correctly with all the props 1`] = `
.c12 {
  font-size: 16px;
  cursor: pointer;
}

.c12 * {
  fill: #616161;
}

.c1 {
  font-size: 14px;
  font-weight: 400;
  color: #616161;
  margin: 0px 8px 0px 0px;
  cursor: default;
}

.c6 {
  font-size: 14px;
  font-weight: 400;
  color: #616161;
  margin: 0px 0px 0px 8px;
  cursor: not-allowed;
}

.c4 {
  grid-area: field;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-left: 0px;
}

.c2 {
  grid-area: label;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  margin: 8px 0px;
  display: grid;
  grid-template-columns: minmax(max-content,25%) minmax(auto,80%);
  grid-template-areas: 'label field' 'random description';
}

.c0 .c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c5 {
  margin: 8px 8px 8px 0px;
  display: inline-grid;
  grid-template-columns: max-content minmax(max-content,25%);
  grid-template-areas: 'field label' 'description random';
}

.c5 .c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c7 {
  position: relative;
  width: 16px;
  height: 16px;
  border: 1px solid #E0E0E0;
  border-radius: 15%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7:focus-within {
  box-shadow: 0 0 0 1pt #012040;
}

.c7 > * {
  width: 100%;
  height: 100%;
  border-radius: 15%;
}

.c10 {
  z-index: 1;
  position: absolute;
  background-color: #ffffff;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 .c11 {
  width: 0;
  height: 0;
  -webkit-transition: width 0.2s ease,height 0.2s ease;
  transition: width 0.2s ease,height 0.2s ease;
}

.c8 {
  opacity: 0;
  z-index: 2;
  position: absolute;
  margin: 0;
  cursor: pointer;
}

.c8:focus {
  outline: none;
}

.c8:disabled {
  cursor: not-allowed;
}

.c8:disabled + .c9 {
  background: #E0E0E0;
}

.c8:checked + .c9 .c11 {
  width: calc(70%);
  height: calc(70%);
}

<div>
  <div
    class="c0"
    id="Cars-checkboxGroup"
  >
    <label
      class="c1 c2"
    >
      Cars
    </label>
    <div
      class="c3 c4"
    >
      <div
        class="c5"
        id="Honda-checkbox"
      >
        <label
          class="c6 c2"
          for="Honda"
        >
          Honda
        </label>
        <div
          class="c7"
          disabled=""
        >
          <input
            class="c8"
            disabled=""
            id="Honda"
            name="honda"
            type="checkbox"
          />
          <div
            class="c9 c10"
          >
            <svg
              class="c11 c12"
              height="1em"
              viewBox="0 0 18 14"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 11.17L1.83 7 .41 8.41 6 14 18 2 16.59.59z"
                fill="none"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="c5"
        id="Hyundai-checkbox"
      >
        <label
          class="c6 c2"
          for="Hyundai"
        >
          Hyundai
        </label>
        <div
          class="c7"
          disabled=""
        >
          <input
            class="c8"
            disabled=""
            id="Hyundai"
            name="hyundai"
            type="checkbox"
          />
          <div
            class="c9 c10"
          >
            <svg
              class="c11 c12"
              height="1em"
              viewBox="0 0 18 14"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 11.17L1.83 7 .41 8.41 6 14 18 2 16.59.59z"
                fill="none"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="c0"
        id="Tata-checkboxGroup"
      >
        <label
          class="c1 c2"
        >
          Tata
        </label>
        <div
          class="c3 c4"
        >
          <div
            class="c5"
            id="Jaguar-checkbox"
          >
            <label
              class="c6 c2"
              for="Jaguar"
            >
              Jaguar
            </label>
            <div
              class="c7"
              disabled=""
            >
              <input
                class="c8"
                disabled=""
                id="Jaguar"
                name="jaguar"
                type="checkbox"
              />
              <div
                class="c9 c10"
              >
                <svg
                  class="c11 c12"
                  height="1em"
                  viewBox="0 0 18 14"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6 11.17L1.83 7 .41 8.41 6 14 18 2 16.59.59z"
                    fill="none"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="c5"
            id="Land Rover-checkbox"
          >
            <label
              class="c6 c2"
              for="Land Rover"
            >
              Land Rover
            </label>
            <div
              class="c7"
              disabled=""
            >
              <input
                class="c8"
                disabled=""
                id="Land Rover"
                name="landRover"
                type="checkbox"
              />
              <div
                class="c9 c10"
              >
                <svg
                  class="c11 c12"
                  height="1em"
                  viewBox="0 0 18 14"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6 11.17L1.83 7 .41 8.41 6 14 18 2 16.59.59z"
                    fill="none"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
